komodo/cli.py

```diff
@@ -25,6 +25,47 @@
 from komodo.yaml_file_types import ReleaseFile, RepositoryFile
 
 
+class KomodoNamespace(argparse.Namespace):
+    """
+    Komodo argument parser namespace
+    """
+
+    pkgs: ReleaseFile
+    repo: RepositoryFile
+    prefix: str
+    release: str
+    tmp: str
+    downloads: str
+    jobs: int
+    download: bool
+    build: bool
+    install: bool
+    dry_run: bool
+    overwrite: bool
+    cmake: str
+    pip: str
+    pyver: Optional[str]
+    workspace: str
+    extra_data_dirs: List[str]
+    postinst: str
+
+
+def profile_time(msg: str) -> Callable[[Callable[..., Any]], Callable[..., Any]]:
+    def decorator(fun: Callable[..., Any]) -> Callable[..., Any]:
+        def wrapper(*args: Any, **kwargs: Any) -> Any:
+            start_time = datetime.datetime.now()
+            res = fun(*args, **kwargs)
+            timings.append(
```

Comment:
where is `timings` coming from?

Reply:
Not sure to tbh. Regardless, you need to specify `global timings` since you modify the content of timings.

Reply:
Upon prototyping, we found out you dont need the global keyword when you are appending to a list; only when you are overwriting the variable in the global scope.